torch/_dynamo/convert_frame.py

```diff
@@ -1131,72 +1100,55 @@ def clean_for_json(d: Dict[str, Any]) -> Dict[str, Any]:
-                config.suppress_errors,
-                config.inline_inbuilt_nn_modules,
-                config.specialize_float,
-                json.dumps(config_dict),
-                True,  # is_forward
-                to_int_ms(remote_fx_graph_cache_get_time),
-                to_int_ms(remote_fx_graph_cache_put_time),
-                start_time_us=start_time_ns // 1000,
-                duration_us=duration_ns // 1000,
-                dynamo_cumulative_compile_time_us=to_int_us(entire_frame_compile_time),
-                aot_autograd_cumulative_compile_time_us=to_int_us(backend_compile_time),
-                inductor_cumulative_compile_time_us=to_int_us(inductor_compile_time),
-                inductor_code_gen_cumulative_compile_time_us=to_int_us(code_gen_time),
-                triton_compile_time_us=None,  # TODO: instrument
-                runtime_cudagraphify_time_us=None,  # TODO: instrument in separate event
-                runtime_triton_autotune_time_us=None,  # TODO: instrument in separate event
-                dynamo_compile_time_before_restart_us=to_int_us(
+            metrics = {
+                "compile_id": str(compile_id),
+                "frame_key": frame_key,
+                "co_name": code.co_name,
+                "co_filename": code.co_filename,
+                "co_firstlineno": code.co_firstlineno,
+                "cache_size": cache_size.num_cache_entries_with_same_id_matched_objs,
+                "accumulated_cache_size": cache_size.num_cache_entries,
+                "guard_count": guard_count,
+                "shape_env_guard_count": shape_env_guard_count,
+                "graph_op_count": graph_op_count,
+                "graph_node_count": graph_node_count,
+                "graph_input_count": graph_input_count,
+                "start_time": start_time_ns / 1e9,
+                "fail_type": fail_type,
+                "fail_reason": fail_reason,
+                "fail_user_frame_filename": fail_user_frame_filename,
+                "fail_user_frame_lineno": fail_user_frame_lineno,
+                "non_compliant_ops": non_compliant_ops,
+                "compliant_custom_ops": compliant_custom_ops,
+                "restart_reasons": restart_reasons,
+                "dynamo_time_before_restart_s": dynamo_time_before_restart,
+                "has_guarded_code": guarded_code is not None,
+                "possibly_missed_reinplacing_opportunities": possibly_missed_reinplacing_opportunities,
+                "structured_logging_overhead_s": structured_logging_overhead_s,
+                "config_suppress_errors": config.suppress_errors,
+                "config_inline_inbuilt_nn_modules": config.inline_inbuilt_nn_modules,
+                "specialize_float": config.specialize_float,
+                "dynamo_config": json.dumps(config_dict),
+                "is_forward": True,
+                "start_time_us": start_time_ns // 1000,
+                "duration_us": duration_ns // 1000,
```

Comment:
Of all the things that should be omitted from manual updates, start time and duration are probably the MOST important to omit. You know what the start/end time are in the decorator, use those values.

Reply:
Agreed, but needs a little refactoring. The exceptions are the same (maybe you comment on those to). If you're ok with it, I'd prefer to do it in a separate PR.

Reply:
added TODO above